vlasovsolver/cpu_trans_map_amr.cpp

```diff
@@ -716,37 +830,129 @@ void getSeedIds(const dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpiGr
    int myRank;
    if (debug) MPI_Comm_rank(MPI_COMM_WORLD,&myRank);
    
-   int neighborhood = getNeighborhood(dimension,1);
-   
-   for(auto celli : localPropagatedCells) {
+   // These neighborhoods now include the AMR addition beyond the regular vlasov stencil
+   int neighborhood = getNeighborhood(dimension,VLASOV_STENCIL_WIDTH);
 
-      auto myIndices = mpiGrid.mapping.get_indices(celli);
-      
-      bool addToSeedIds = P::transShortPencils;
+#pragma parallel for
+   for (uint i=0; i<localPropagatedCells.size(); i++) {
+      CellID celli = localPropagatedCells[i];
+
+      bool addToSeedIds = P::amrTransShortPencils;
       if (addToSeedIds) {
+#pragma omp critical
          seedIds.push_back(celli);
          continue;
       }
-
+      auto myIndices = mpiGrid.mapping.get_indices(celli);
+      int myRefLevel;
+
+      /* -----------------------------------------
+         | A |   | B |   |_|_|_|_|   |   | C |   |
+         |   |   |   |   | | | | |   |   |   |   |
+         -----------------------------------------
+         For optimal pencil generation, we need seedids at A, B, and C.
+         A Is triggered in the first if-clause. Pencils starting from B
+         will be split (but won't cause A to split), and pencils from
+         C will be able to remain again un-split. These checks need to be done
+         only if we aren't already at the maximum refinement level.
+      */
```

Comment:
This is awesome. Really good optimization, and elegant in its implementation, too!